modules/transformation.py

```diff
@@ -0,0 +1,500 @@
+            'renderStart': page.get('_render'),
+            'fullyLoaded': page.get('_fullyLoaded'),
+            'visualComplete': page.get('_visualComplete'),
+            'onLoad':
+                page.get('_docTime')
+                if page.get('_docTime') != 0
+                else max(page.get('_visualComplete'), page.get('_fullyLoaded')),
+            'gzipTotal': page.get('_gzip_total'),
+            'gzipSavings': page.get('_gzip_savings'),
+            'numDomElements': page.get('_domElements'),
+            'onContentLoaded': page.get('_domContentLoadedEventStart'),
+            'cdn': page.get('_base_page_cdn'),
+            'SpeedIndex': page.get('_SpeedIndex'),
+            'PageSpeed': page.get('_pageSpeed', {}).get('score'),
+            '_connections': page.get('_connections'),
+            '_adult_site': page.get('_adult_site', False),
+            'avg_dom_depth': page.get('_avg_dom_depth'),
+            'doctype': page.get('_doctype'),
+            'document_height': page['_document_height'] if page.get('_document_height') and int(
+                page['_document_height']) > 0 else 0,
+            'document_width': page['_document_width'] if page.get('_document_width') and int(
+                page['_document_width']) > 0 else 0,
+            'localstorage_size': page['_localstorage_size'] if page.get('_localstorage_size') and int(
+                page['_localstorage_size']) > 0 else 0,
+            'sessionstorage_size': page['_sessionstorage_size'] if page.get('_sessionstorage_size') and int(
+                page['_sessionstorage_size']) > 0 else 0,
+            'meta_viewport': page.get('_meta_viewport'),
+            'num_iframes': page.get('_num_iframes'),
+            'num_scripts': page.get('_num_scripts'),
+            'num_scripts_sync': page.get('_num_scripts_sync'),
+            'num_scripts_async': page.get('_num_scripts_async'),
+            'usertiming': page.get('_usertiming')
+        }
+
+    @staticmethod
+    def aggregate_stats(entries, page_id, first_url, first_html_url, status_info):
+        # CVSNO - move this error checking to the point before this function is called
+        if not first_url:
+            logging.error("ERROR($gPagesTable pageid: {}}): no first URL found.", page_id)
+            return
+        if not first_html_url:
+            logging.error("ERROR($gPagesTable pageid: {}): no first HTML URL found.", page_id)
+            return
+
+        # initialize variables for counting the page's stats
+        bytes_total = 0
+        req_total = 0
+        size = {}
+        count = {}
+
+        # This is a list of all mime types AND file formats that we care about.
+        typs = ["css", "image", "script", "html", "font", "other", "audio", "video", "text", "xml", "gif", "jpg", "png",
+                "webp", "svg", "ico", "flash", "swf", "mp4", "flv", "f4v"]
+        # initialize the hashes
+        for typ in typs:
+            size[typ] = 0
+            count[typ] = 0
```

Comment:
Should we use types/type tather than typs/typ?

We recently added `.avif`, `.jxl`, `.heic`, and `.heif` so should we add them here? They should be captured as images but still m would be good for consistency. Also wonder if we should add `.json`?